Library-Light-Twin-DSD2023-MDU-POLIMI/digital-twin-dsd-backend
# Release beta

Release Message: PR Merge from Dev to Main
🚀 New Features in GraphQL API

We're excited to announce a significant update to our GraphQL API, now merged from the development branch to the main branch. This update introduces a comprehensive set of new queries and mutations, enhancing our data handling capabilities and providing more flexibility for our users. Here's what's new:

**Enhanced Queries**:
```
Lighting Assets: Retrieve individual or multiple lighting assets with advanced filtering and pagination.
Time Series Data: Fetch time series data for lighting assets within specified time ranges, including threshold-based queries.
Average Data Calculation: Get average data for lighting assets over a specified time period.
Metric Handling: Access individual metrics or a list of all metrics.
Work Order Management: Query individual or multiple work orders.
Robust Mutations:
Asset Management: Add, update, or remove lighting assets.
Work Order Operations: Create, update, or remove work orders.
Measurement and Metrics: Add measurements to lighting assets and manage metric data, including adding, updating, or removing metrics.
```


**Testing Instructions**
To ensure the reliability of these new features, we have included comprehensive tests. Please use `npx jest` to run these tests. We recommend executing one test at a time for optimal tracking of individual feature performance.

We're committed to continually improving our API and appreciate your feedback on these new features. Happy coding! 🌟
